pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,32 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+			today := time.Now().Format("2006-01-02")
+			if secret.GetAnnotations()[LastUsedAnnotationKey] != today || secret.GetLabels()[InUseLabelKey] != today {
+				if secret.GetAnnotations() == nil {
+					secret.SetAnnotations(map[string]string{})
+				}
+				secret.GetAnnotations()[LastUsedAnnotationKey] = today
+				if secret.GetLabels() == nil {
+					secret.SetLabels(map[string]string{})
+				}
+				secret.GetLabels()[InUseLabelKey] = today
+
+				patchContent, err := json.Marshal(applyv1.ObjectMeta().WithLabels(map[string]string{LastUsedAnnotationKey: today}).WithAnnotations(map[string]string{InUseLabelKey: today}))
+				if err != nil {
+					klog.Errorf("Failed to marshall legacy service account token tracking annotations, err: %v", err)
+				}
+				if _, err := v.secretsWriter.Secrets(namespace).Patch(ctx, secret.Name, types.MergePatchType, patchContent, metav1.PatchOptions{}); err != nil {
+					klog.Errorf("Failed to annotate legacy service account token secret with last-used, err: %v", err)
+				}
+				if _, err := v.secretsWriter.Secrets(namespace).Update(ctx, secret, metav1.UpdateOptions{}); err != nil {
+					klog.Errorf("Failed to annotate legacy service account token secret with last-used, err: %v", err)
+				}
```

Comment:
drop this, the patch should be sufficient

Reply:
forgot to delete this.